File: src/main/java/com/github/cybellereaper/wizpets/core/config/PluginConfig.java
package com.github.cybellereaper.wizpets.core.config;

import java.util.Objects;
import org.bukkit.configuration.file.FileConfiguration;

public final class PluginConfig {
    private final FileConfiguration configuration;

    public PluginConfig(FileConfiguration configuration) {
        this.configuration = Objects.requireNonNull(configuration, "configuration");
    }

    public boolean isAutoSummon() {
        return configuration.getBoolean("autosummon", true);
    }

    public String getMountUnlockMessage() {
        return configuration.getString("messages.mountUnlock", "§bYour bond deepens, allowing you to ride your pet!");
    }

    public String getFlightUnlockMessage() {
        return configuration.getString("messages.flightUnlock", "§3Your pet lifts you into the skies!");
    }

    public String getDefaultPetName() {
        return configuration.getString("defaults.displayName", "Wisp Familiar");
    }
}

File: src/main/kotlin/com/github/cybellereaper/wizpets/core/config/PluginConfig.kt
package com.github.cybellereaper.wizpets.core.config

import org.bukkit.configuration.file.FileConfiguration

class PluginConfig(private val config: FileConfiguration) {
    val autoSummon: Boolean get() = config.getBoolean("autosummon", true)
    val mountUnlockMessage: String get() = config.getString("messages.mountUnlock") ?: "§bYour bond deepens, allowing you to ride your pet!"
    val flightUnlockMessage: String get() = config.getString("messages.flightUnlock") ?: "§3Your pet lifts you into the skies!"
    val defaultPetName: String get() = config.getString("defaults.displayName") ?: "Wisp Familiar"
}
